lib/class-wp-rest-annotations-controller.php

```diff
@@ -0,0 +1,1065 @@
+<?php
```

Comment:
Hey Jason, isn't there a way to break down this humongous file into smaller modules? There is so much going on in here, it will be hard to maintain imo. Same as the utils class above. The separation of what is covered by each is not clear to me either. Maybe start by extracting the permission checks for a start, then gradually organise things maybe per CRUD operation., identify reusable patterns, etc. 

This is obviously a big task, and by all means kudos for writing so much code. Just a little bit more organisation will go a long way.

Reply:
@chriskmnds Thanks for the review 😄 

> lib/class-wp-rest-annotations-controller.php
> break down this humongous file into smaller modules? 

**WP_REST_Annotations_Controller** extends [WP_REST_Comments_Controller](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php) in core. There's a [nice article in the handbook](https://developer.wordpress.org/rest-api/extending-the-rest-api/controller-classes/) that summarizes the infrastructure.

While I'm not opposed to separating areas of concern, I think it's important to follow the precedent that's been set and to maintain a consistent pattern.

---

> class-wp-annotation-utils.php
> extracting the permission checks for a start

I agree. Pulling the permissions out of there seems like a great idea.